src/ipc/format-usb-drive.ts

```diff
@@ -0,0 +1,100 @@
+import makeDebug from 'debug';
+import { IpcMain, IpcMainInvokeEvent } from 'electron';
+import { join } from 'path';
+import exec, { ExecError } from '../utils/exec';
+
+const debug = makeDebug('kiosk-browser:format-usb-drive');
+
+export const channel = 'formatUsbDrive';
+
+const PARTITION_FORMAT_CODES: Record<KioskBrowser.UsbFormat, string> = {
+  fat32: 'c',
+  exfat: '7',
+};
+
+const FS_MAX_NAME_LENGTH: Record<KioskBrowser.UsbFormat, number> = {
+  fat32: 11,
+  exfat: 15,
+};
+
+const MKFS_ARGS: Record<KioskBrowser.UsbFormat, string[]> = {
+  fat32: ['mkfs.fat', '-F', '32'],
+  exfat: ['mkfs.exfat'],
+};
```

Comment:
I'm starting to use `Record` types to track maps from known values to something else. It's more compact than `switch` statements, more readable than ternary or if statements, and more portable. And it's still type-safe.